pkg/services/ngalert/api/api_alertmanager.go

```diff
@@ -45,8 +45,15 @@ func (srv AlertmanagerSrv) RouteCreateSilence(c *models.ReqContext, postableSile
 }
 
 func (srv AlertmanagerSrv) RouteDeleteAlertingConfig(c *models.ReqContext) response.Response {
-	// not implemented
-	return NotImplementedResp
+	if !c.HasUserRole(models.ROLE_EDITOR) {
+		return ErrResp(http.StatusForbidden, errors.New("permission denied"), "")
+	}
+	if err := srv.am.SaveAndApplyDefaultConfig(); err != nil {
+		srv.log.Error("unable to save and apply default alertmanager configuration", "err", err)
+		return ErrResp(http.StatusBadRequest, err, "failed to save and apply default Alertmanager configuration")
```

Comment:
nit: does a 400 or a 500 make more sense here? Could be a reason other than a bad request that we failed to save the config

Reply:
I will change it; `BadRequest` does not make any sense in this case.